drivers/timer/renesas_rx_cmt.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2024 Renesas Electronics Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "zephyr/sys_clock.h"
+#include <zephyr/arch/cpu.h>
+#include <zephyr/init.h>
+#include <soc.h>
+#include <zephyr/drivers/timer/system_timer.h>
+#include <zephyr/irq.h>
+#include <platform.h>
+
+#define DT_DRV_COMPAT renesas_rx_timer_cmt
+
+#define CMT_NODE  DT_NODELABEL(cmt)
+#define CMT0_NODE DT_NODELABEL(cmt0)
+
+#define CMT_CMSTR0_ADDR DT_REG_ADDR_BY_NAME(CMT_NODE, CMSTR0)
+#define CMT0_CMCR_ADDR  DT_REG_ADDR_BY_NAME(CMT0_NODE, CMCR)
+#define CMT0_CMCNT_ADDR DT_REG_ADDR_BY_NAME(CMT0_NODE, CMCNT)
+#define CMT0_CMCOR_ADDR DT_REG_ADDR_BY_NAME(CMT0_NODE, CMCOR)
+
+#define ICU_NODE DT_NODELABEL(icu)
+
+#define ICU_IR_ADDR  DT_REG_ADDR_BY_NAME(ICU_NODE, IR)
+#define ICU_IER_ADDR DT_REG_ADDR_BY_NAME(ICU_NODE, IER)
+#define ICU_IPR_ADDR DT_REG_ADDR_BY_NAME(ICU_NODE, IPR)
+#define ICU_FIR_ADDR DT_REG_ADDR_BY_NAME(ICU_NODE, FIR)
+
+#define COUNTER_MAX   0x0000ffff
+#define TIMER_STOPPED 0xff000000
+
+/* The following macro values will be changed later to be calculated using the clock control driver.
+ * This macro is the CMT count clock. (=PCLKB/8)
+ */
+#define CYCLES_PER_SEC  (CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC)
+#define TICKS_PER_SEC   (CONFIG_SYS_CLOCK_TICKS_PER_SEC)
+#define CYCLES_PER_TICK (CYCLES_PER_SEC / TICKS_PER_SEC)
+#define MAX_TICKS       ((COUNTER_MAX / CYCLES_PER_TICK) - 1)
+
+#if (IS_ENABLED(CONFIG_TICKLESS_KERNEL))
+#warning Warning: Tickless mode is not yet implemented.
+#endif
+
+typedef uint32_t cycle_t;
+static cycle_t cycle_count;
+
+/* clock_cycles_per_tick is calculated once in
+ * sys_clock_driver_init() to avoid repeated division of sub_clk_freq_hz
+ * and TICKS_PER_SEC. This is justified as clock frequency should not
+ * change during operation without undesired side effects on peripheral
+ * devices.
+ */
+static uint16_t clock_cycles_per_tick;
+
+static uint32_t last_load;
+
+uint32_t sys_clock_cycle_get_32(void)
+{
+	uint32_t ret = cycle_count;
+
+	volatile uint16_t *cmt0_cmcnt = (uint16_t *)CMT0_CMCNT_ADDR; /* CMCNT */
+	ret += (uint32_t)(*cmt0_cmcnt);
+
+	return ret;
+}
+
+uint32_t sys_clock_elapsed(void)
+{
+	if (!IS_ENABLED(CONFIG_TICKLESS_KERNEL)) {
+		/* Always return 0 for tickful operation */
+		return 0;
+	}
+
+	volatile uint16_t *cmt0_cmcnt = (uint16_t *)CMT0_CMCNT_ADDR; /* CMCNT */
+	return ((uint32_t)((*cmt0_cmcnt) / clock_cycles_per_tick));
+}
+
+static void cmt0_isr(void)
+{
+	k_ticks_t dticks;
+
+	cycle_count += CYCLES_PER_TICK;
+	dticks = 1;
+	sys_clock_announce(dticks);
+}
+
+#define CMT0_IRQ_NUM DT_IRQ_BY_NAME(CMT0_NODE, cmi, irq)
```

Comment:
move defines to top of file with other defines

Reply:
It's fixed